src/neuroconv/datainterfaces/behavior/video/internalvideointerface.py

```diff
@@ -0,0 +1,345 @@
+import warnings
+from copy import deepcopy
+from pathlib import Path
+from typing import Literal, Optional
+
+import numpy as np
+import psutil
+from hdmf.data_utils import DataChunkIterator
+from pydantic import FilePath, validate_call
+from pynwb import NWBFile
+from pynwb.image import ImageSeries
+from tqdm import tqdm
+
+from .video_utils import VideoCaptureContext
+from ....basedatainterface import BaseDataInterface
+from ....tools import get_package
+from ....tools.nwb_helpers import get_module
+from ....utils import dict_deep_update, get_base_schema, get_schema_from_hdmf_class
+from ....utils.str_utils import human_readable_size
+
+
+class InternalVideoInterface(BaseDataInterface):
+    """Data interface for writing videos as ImageSeries."""
+
+    display_name = "Video"
+    keywords = ("movie", "natural behavior", "tracking")
+    associated_suffixes = (".mp4", ".avi", ".wmv", ".mov", ".flx", ".mkv")
+    # Other suffixes, while they can be opened by OpenCV, are not supported by DANDI so should probably not list here
+    info = "Interface for handling standard video file formats."
+
+    @validate_call
+    def __init__(
+        self,
+        file_path: FilePath,
+        verbose: bool = False,
+        *,
+        video_name: str = "Video1",
+    ):
+        """
+        Create the interface for writing videos as ImageSeries.
+
+        Parameters
+        ----------
+        file_paths : list of FilePaths
+            Many video storage formats segment a sequence of videos over the course of the experiment.
+            Pass the file paths for this videos as a list in sorted, consecutive order.
+        metadata_key_name : str, optional
+            The key used to identify this video data within the overall experiment metadata.
+            Defaults to "Videos".
+
+            This key is essential when multiple video streams are present in a single experiment.
+            The associated metadata should be a list of dictionaries, with each dictionary
+            containing metadata for a specific video segment:
+
+            ```
+            metadata["Behavior"][metadata_key_name] = [
```

Comment:
I think the docstring does not represent the new naming that you have.

So, if we follow https://github.com/catalystneuro/neuroconv/issues/269 this would change to a dict of vide metadata instead of a list, right?